### ISSUE TITLE
Summary page of batch search should load faster

### ISSUE THREAD

**Issue by nikiforosk** [opentargets/webapp#69]
# Problem description

While testing the first iteration of batch search (pull request #16) [several participants complained that it takes a long time to load](https://docs.google.com/document/d/1ZDPWqSHhw4Kl_cOqBkTTaWNOG5VikrOLapeA2NCGz5w/edit#heading=h.728drs263vag).

# Testing the fix
- How can we tell reliably that the page is loading faster @emepyc ?

**Comment by emepyc** [comment]
@nikiforosk , the metric is people not complaining about the loading speed? :-)
This issue shouldn't live in the webapp repo though. The problem comes from the rest api and the current data model

**Comment by nikiforosk** [comment]
For speed in particular no news isn't generally good news. People expect modern websites to be fast and won't necessarily complain to us if ours isn't fast enough. 

Conventional UX wisdom suggests a loading speed of up to 1 sec as acceptable:
https://www.nngroup.com/articles/website-response-times/

After that people abandon a site at an alarmingly increasing rate see e.g.:
https://blog.kissmetrics.com/loading-time/

Another metric to use is the Exit rate (especially if we can correlate it with loading speed). But this will be applicable only after the fact and will probably confirm what we expect to happen (i.e. more people dropping out as loading time increases).

**Comment by emepyc** [comment]
I think that bit of conventional ux wisdom should be taken with a grain of salt in domain-specific web applications.
Of course the faster the better, but it is unrealistically to think that we will go under 1s for the enrichment analysis (nor any other web application doing the same on the same amount of data).

**Comment by apierleoni** [comment]
I guess we should aim to a speed that is fast enough for our users. Remember that they are used to systems in which they make a query and come back after hours to get the response. 
We are not an ecommerce website we, don't have to worry about conversions and missed sales.

**Comment by nikiforosk** [comment]
I think increasing the loading speed should be our #1 priority. People are not comparing us with the systems that take hours but with Google... 

We don't worry about missed sales but we should worry about users dropping out. I understand that this is challenge but let's not be complacent.

**Comment by emepyc** [comment]
This has been already done in 2.1
